crates/ironrdp-pdu/src/rdp/suppress_output.rs

```diff
@@ -0,0 +1,57 @@
+use std::io;
+
+use byteorder::{ReadBytesExt as _, WriteBytesExt as _};
+use num_derive::{FromPrimitive, ToPrimitive};
+use num_traits::{FromPrimitive, ToPrimitive};
+
+use crate::{geometry::InclusiveRectangle, PduParsing};
+
+#[repr(u8)]
+#[derive(Debug, PartialEq, Eq, FromPrimitive, ToPrimitive, Clone)]
+pub enum AllowDisplayUpdatesType {
+    SuppressDisplayUpdates = 0x00,
+    AllowDisplayUpdates = 0x01,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SuppressOutputPdu {
+    allow_display_updates: AllowDisplayUpdatesType,
+    desktop_rect: Option<InclusiveRectangle>,
+}
+
+impl PduParsing for SuppressOutputPdu {
+    type Error = io::Error;
+
+    fn from_buffer(mut stream: impl io::Read) -> Result<Self, Self::Error> {
+        let allow_display_updates = AllowDisplayUpdatesType::from_u8(stream.read_u8()?)
+            .ok_or_else(|| io::Error::new(io::ErrorKind::InvalidData, "Invalid display update type"))?;
+        let _padding = stream.read_u8()?; // padding
+        let _padding = stream.read_u8()?; // padding
+        let _padding = stream.read_u8()?; // padding
+        let desktop_rect = if allow_display_updates == AllowDisplayUpdatesType::AllowDisplayUpdates {
+            Some(InclusiveRectangle::from_buffer(&mut stream)?)
+        } else {
+            None
+        };
+        Ok(Self {
+            allow_display_updates,
+            desktop_rect,
+        })
+    }
+
+    fn to_buffer(&self, mut stream: impl io::Write) -> Result<(), Self::Error> {
+        stream.write_u8(self.allow_display_updates.to_u8().unwrap())?;
+        stream.write_u8(0)?; // padding
+        stream.write_u8(0)?; // padding
+        stream.write_u8(0)?; // padding
```

Comment:
Nitpick: we have the `write_padding!` macro specifically for this

Rationale: it communicates the intent by its name and always ensure padding is zeroed

Reply:
ah.. it doesn't work with streams..

Reply:
Oh right, this code is implementing the legacy trait (I assume because the rest of the module is too). You can forget about my comment